src/shared/ui/SketchCanvas/SketchCanvas.tsx

```diff
@@ -166,10 +170,21 @@ const SketchCanvas = forwardRef<SketchCanvasRef, Props>((props, ref) => {
   const preprocessedTouchHandler = (
     touchInfoMatrix: Array<Array<TouchInfo>>,
   ) => {
-    const matrix = new TouchInfoMatrix(touchInfoMatrix);
-    const updatedTouchInfoMatrix = matrix.filterOutMultiTouches();
+    const { type, id } = touchInfoMatrix[0][0];
 
-    touchHandler(updatedTouchInfoMatrix);
+    if (type === TouchType.Start && currentTouchIdRef.current === null) {
+      currentTouchIdRef.current = id;
+    }
+
+    let matrix = new TouchInfoMatrix(touchInfoMatrix);
```

Comment:
Can we keep the instance of TouchInfoMatrix in the ref and get it via function: 
`getMatrix = () => matrixRef.current`
Also:
Remove the currentTouchIdRef.current and add a similar property into the TouchInfoMatrix. So that, we'd set the currentTouchId directly into the instance of the class.

Reply:
I respect your approach and it's something I thought about but, in my opinion, it will not be beneficial for us to add a matrix to Ref:
1. The matrix value is unique upon Canvas's `onTouch` calls and its value is used once - in `touchHandler` function. After that, the array is garbage-collected.
2. The only thing that is crucial to memorize is the Touch ID. The matrix's `TouchInfo` objects may not have the Touch ID that is stored in `currentTouchIdRef`.

Theoretically, we could create a `TouchInfoManager` class that could do what you have described. However, those two points above are still valid even in this case, and the class would have to have many communications with the outer scope (e.g. through callbacks) because we need to have references to `currentTouchIdRef` and `touchHandler`